hr_employee_calendar_planning/models/hr_employee.py

```diff
@@ -64,6 +64,11 @@ def _regenerate_calendar(self):
                 )
                 .id
             )
+            # We need to ensure that the related gets the proper autogenerated
+            # calendar when we create the employees. Otherwise it remains as the company
+            # default one
+            if self.resource_id.calendar_id != self.resource_calendar_id:
+                self.resource_id.calendar_id = self.resource_calendar_id
```

Comment:
Now that I see the code, should we assign `self.resource_id.calendar_id` in the `create` command instead?

Reply:
Yes, it was what I was doing in the first commit although inderectly triggering the relation setting

Reply:
I meant here:

https://github.com/OCA/hr/pull/1003/files#diff-edb2d081d492df756c64e4a9cf821b691b64c7f3cf8b94b3947f151de9810a2dR54

Chat me if further clarification.

Reply:
Understood and changed!